### PR TITLE
Simplify React frontend

### PR DESCRIPTION
**Pull request by therobinkim**
This is a follow up to #35.

The big changes are:
* Upgrade from React 16.2 to React 16.13.1 to take advantage of the new lifecycle methods like `static getDerivedPropsFromState()` (and not use soon-to-be-deprecated lifecycle methods like `componentWillReceiveProps()`). https://medium.com/@baphemot/whats-new-in-react-16-3-d2c9b7b6193b
* `<BrowserRouter />` is being used instead of `<HashRouter />` from React Router 4 now.
* You can ONLY see a collection or ONLY see the details of a moji, but not BOTH at the same time. Why? Just to keep the code & cognitive overhead easier/simpler.
* `<MojiDetails />` component has been renamed to `<Moji />`.
* `<Browse />` component has been created, and it determines whether `<Collection />` component or `<Moji />` component is rendered. (It will soon have the option to display the upcoming `<Offer />` component.

What's not changed:
* `live-server` is still used bc it's working fine for me, and I didn't want to slow myself down during this curriculum development phase. Happy to get rid of it later.

### CODE REVIEW

**Comment by therobinkim** [comment]
@delventhalz `collection_cats1` and `moji_cat1` are the intended addresses for this PR.. I even did a `git checkout` to the commit I'm making this PR for and `rm -rf node_modules && npm install`, and it's working for me. 🤔

**Comment by therobinkim** [comment]
@delventhalz ah, I think I might've figured out why

1. did a fresh `git clone` operation in a separate directory
2. ran `docker-compose up` from `/code/part-two/`
3. ran `npm install` and `npm run build:dev` from `/code/part-two/client/`
4. loaded up http://localhost:8000/, and interacting w/ the webpage from there, which seems to work for me

HOWEVER, reloading the web page when at anything but localhost:8000 seems to result in a 404 bc Apache isn't configured to serve `index.html` for unknown routes (?). Soooo I guess back to React Router's `<HashRouter />` (instead of `<BrowserRouter />`) unless we can config Apache to work for this SPA?

File: code/part-two/client/source/Browse.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    this.state = {
+      address: null,
+      type: null,
+      collection: null,
+      moji: null
+    };
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    console.log('GETDERIVEDSTATEFROMPROPS: <Browse />');
```
[the remark this answers is not in the document]
Comment (therobinkim) [reply code/part-two/client/source/Browse.jsx]: Yes, but is it OK to keep them in for now? I discovered an infinite loop of updates that I wouldn't have discovered if I hadn't kept these console.logs in for now. (I guess I could also just remove them and keep them on my end w/o committing them, if you'd like.)

**Comment by delventhalz** [comment]
@therobinkim Ah, okay. Hrmm. I've always used Apache in the past with hash routes. I haven't configured Apache to serve a SPA properly without the `#`, but it should be possible. No idea how much of a pain it would be though . . .

Alright I found this page:

And I added this to line 16 of client/Dockerfile:
```
LoadModule rewrite_module modules/mod_rewrite.so\n\
RewriteEngine On\n\
RewriteRule ^ /index.html\n\
\n\
```

This got rid of the 404's, but I'm still not seeing anything display on the page other than the search bar still. I will try to debug further tonight.

**Comment by delventhalz** [comment]
Ah, I forgot to `npm run build:dev`!

Okay, the Apache fix works. If you could add a commit that adds the above code to the Dockerfile, I am satisfied.

**Comment by therobinkim** [comment]
@delventhalz Added new lines for client's Dockerfile, but the new changes didn't work for me. :( When I'm at http://127.0.0.1:8000/browse/collection_cats1 or http://localhost:8000/browse/collection_cats1 and refresh, I get a 404. I've committed my work and updated my PR so you can see what I tried, in case I updated the wrong file or something. Any suggestions on how to proceed? In the meantime, I'll continue prepping my next PR.

image

Some of the logs from my `docker-compose up` session:
> client         | 172.18.0.1 - - [12/Apr/2018:19:19:49 +0000] "GET / HTTP/1.1" 200 163
> client         | 172.18.0.1 - - [12/Apr/2018:19:19:49 +0000] "GET /bundle.js HTTP/1.1" 200 949752
> client         | 172.18.0.1 - - [12/Apr/2018:19:19:55 +0000] "GET /browse/collection_cats1 HTTP/1.1" 404 221
> client         | 172.18.0.1 - - [12/Apr/2018:19:20:39 +0000] "GET / HTTP/1.1" 200 163
> client         | 172.18.0.1 - - [12/Apr/2018:19:20:39 +0000] "GET /bundle.js HTTP/1.1" 200 949752

**Comment by delventhalz** [comment]
@therobinkim You have to `docker-compose up --build` if you want to rebuild the docker images. I tested it and it works fine if you rebuild. Approved.

**Pull request merged**